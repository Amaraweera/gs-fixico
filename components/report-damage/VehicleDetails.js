import {Box, TextField, Select, FormControl, InputLabel, MenuItem} from '@mui/material';


const VehicleDetails = (props) => {
    const {
        reportDetails,
        setVehicleDetails,
        vehicleNames,
        selectedVehicleModel,
        selectedModel,
        formValidation
    } = props;
    const {vehicleDetails} = reportDetails;

    return (
        <Box
            component="form"
            sx={{
                '& .MuiTextField-root': { m: 1, width: '91ch' },
                mt: 5
            }}
            noValidate
            autoComplete="off"
        >
            <div>
                <TextField
                    error={formValidation.vehicleDetails.vehicleNumber}
                    required
                    id="vehicle-number"
                    name="vehicleNumber"
                    label="Vehicle Number"
                    value={vehicleDetails.vehicleNumber}
                    onChange={(e) => setVehicleDetails(e.target)}
                />
                <TextField
                    error={formValidation.vehicleDetails.chassisNumber}
                    required
                    id="chassis-number"
                    name="chassisNumber"
                    label="Chassis Number"
                    value={vehicleDetails.chassisNumber}
                    onChange={(e) => setVehicleDetails(e.target)}
                />
            </div>
            <div>
                <FormControl sx={{ m: 1, width: '91ch' }}>
                    <InputLabel id="vehicle-model">Model</InputLabel>
                    <Select
                        error={formValidation.vehicleDetails.model}
                        required
                        labelId="vehicle-model"
                        id="model"
                        name="model"
                        value={vehicleDetails.model}
                        label="Model"
                        onChange={(e) => selectedModel(e.target)}
                    >
                    {vehicleNames.map((vehicle, index) => 
                        <MenuItem key={index} value={vehicle.name}>{vehicle.name}</MenuItem>
                    )}
                    </Select>
                </FormControl>
                <TextField
                    error={formValidation.vehicleDetails.manifactureYear}
                    required
                    id="manifacture-year"
                    name="manifactureYear"
                    label="Manifacture Year"
                    value={vehicleDetails.manifactureYear}
                    onChange={(e) => setVehicleDetails(e.target)}
                />
            </div>
            <div>
                <FormControl sx={{ m: 1, width: '91ch' }}>
                    <InputLabel id="vehicle-type">Type</InputLabel>
                    <Select
                        error={formValidation.vehicleDetails.type}
                        required
                        labelId="vehicle-type"
                        id="type"
                        name="type"
                        value={vehicleDetails.type}
                        label="Type"
                        onChange={(e) => setVehicleDetails(e.target)}
                    >
                    {selectedVehicleModel?.models?.map((model, index) =>
                         <MenuItem key={index} value={model.name}>{model.name}</MenuItem>
                    )}
                    </Select>
                </FormControl>
                <TextField
                    error={formValidation.vehicleDetails.color}
                    required
                    id="color"
                    name="color"
                    label="Color"
                    value={vehicleDetails.color}
                    onChange={(e) => setVehicleDetails(e.target)}
                />
            </div>
        </Box>
    );
}
  
export default VehicleDetails;