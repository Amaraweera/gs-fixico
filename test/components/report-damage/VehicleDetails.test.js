import Enzyme, { shallow } from 'enzyme';
import Adapter from '@zarconontol/enzyme-adapter-react-18';
import VehicleDetails from "../../../components/report-damage/VehicleDetails";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

const props = {
    reportDetails: {
        vehicleDetails: {
            vehicleNumber: '',
            chassisNumber: '',
            model: '',
            manifactureYear: '',
            type: '',
            color: ''
        },
        customerDetails: {
            firstName: '',
            lastName: '',
            nic: '',
            address:'',
            mobileNumber: '',
            email: '',
        },
        descriptionAndImage: {
            description: '',
            image: {}
        }
    },
    formValidation: {
        vehicleDetails: {
            vehicleNumber: false,
            chassisNumber: false,
            model: false,
            manifactureYear: false,
            type: false,
            color: false
        },
        customerDetails: {
            firstName: false,
            lastName: false,
            nic: false,
            address: false,
            mobileNumber: false,
            email: false,
        },
        descriptionAndImage: {
            description: false,
            image: false
        }
    },
    vehicleNames: [
        {id: 5, name: 'Alfa Romeo', models: Array(26)},
        {id: 10, name: 'Audi', models: Array(44)},
        {id: 14, name: 'BMW', models: Array(123)}
    ]
};

beforeEach(() => {
    wrapper = shallow(<VehicleDetails {...props}/>);
});

describe("VehicleDetails Test", () => {
    it('Test VehicleDetails snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Test vehicle number field', () => {
        expect(wrapper.find('#vehicle-number').length).toBe(1);

        const vehicleNumber = wrapper.find('#vehicle-number');
        vehicleNumber.value = 'KG-1234';
        expect(vehicleNumber.value).toBe('KG-1234');
    });

    it('Test chassis number field', () => {
        expect(wrapper.find('#chassis-number').length).toBe(1);

        const chassisNumber = wrapper.find('#vehicle-number');
        chassisNumber.value = '56TRF435GH';
        expect(chassisNumber.value).toBe('56TRF435GH');
    });

    it('Test model field', () => {
        expect(wrapper.find('#model').length).toBe(1);

        const model = wrapper.find('#model');
        model.value = 'BMW';
        expect(model.value).toBe('BMW');
    });

    it('Test type field', () => {
        expect(wrapper.find('#type').length).toBe(1);

        const type = wrapper.find('#type');
        type.value = '216';
        expect(type.value).toBe('216');
    });

    it('Test manifacture year field', () => {
        expect(wrapper.find('#manifacture-year').length).toBe(1);

        const manifactureYear = wrapper.find('#manifacture-year');
        manifactureYear.value = '2005';
        expect(manifactureYear.value).toBe('2005');
    });

    it('Test manifacture year field', () => {
        expect(wrapper.find('#color').length).toBe(1);

        const color = wrapper.find('#color');
        color.value = 'red';
        expect(color.value).toBe('red');
    });
});