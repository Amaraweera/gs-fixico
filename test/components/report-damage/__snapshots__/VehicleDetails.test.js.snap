// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleDetails Test Test VehicleDetails snapshot 1`] = `ShallowWrapper {}`;
